airflow/jobs/scheduler_job.py

```diff
@@ -924,6 +924,7 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
             .filter(not_(DM.is_paused))
             .filter(TI.state == State.SCHEDULED)
             .options(selectinload('dag_model'))
+            .order_by(-TI.priority_weight, TI.execution_date)
```

Comment:
Can you add a test case for this in https://github.com/apache/airflow/blob/master/tests/jobs/test_scheduler_job.py please to avoid regression

Reply:
Done, although I am not sure if the build failure I now get is related to my test or not.

Reply:
@kaxil ideas on what is failing? I see the test_scheduler_job (the one I added my test to) run, and I see dots after it, so I'd think it is passing.

Reply:
How do we determine if the build is failing from master vs the new test?

Reply:
@matt-land Checking the status of CI on the first commit might help to determine that